shared/middleware/net_sim_api.rb

```diff
@@ -79,7 +66,6 @@ def has_json_utf8_headers(request)
     table = get_table(shard_id, table_name)
     int_id = id.to_i
     table.delete(int_id)
```

Comment:
Could you do some very basic coalescing of refreshes on the client so we don't hammer the server? e.g. wait a very short time before doing the refresh to allow other invalidations to trickle in, all of which can be serviced by a single hit per table.